streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala

```diff
@@ -241,6 +242,7 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
 
   /** Generate jobs and perform checkpoint for the given `time`.  */
   private def generateJobs(time: Time) {
+    listenerBus.post(StreamingListenerBatchGenerateStarted(clock.getTimeMillis()))
```

Comment:
The StreamingListenerBatchGenerateStarted and StreamingListenerBatchGenerateCompleted should report the batchTime information as well. If not the end-user will not be able to correlate this processingTime to the relevant batch.